internal/controller/datadogagent/override/global.go

```diff
@@ -137,6 +137,14 @@ func applyGlobalSettings(logger logr.Logger, manager feature.PodTemplateManagers
 		}
 	}
 
+	// Env is a list of custom global variables that are set across all agents.
+	if config.Env != nil {
+		logger.Info(fmt.Sprintf("adding custom env var %s", config.Env))
```

Comment:
```suggestion
```

```suggestion
		logger.V(1).Info(fmt.Sprintf("Adding custom environment variable %s", config.Env))
```
Can we remove this log or set it log to `debug` instead of `info` ? It will be sent a lot (for instance, if using all 3 components, 3 times every reconciliation). The formatting could also be improved slightly, here is how it looks when using: 
```yaml
global:
  env:
  - name: foo
    value: bar
```

```json
{"level":"INFO","ts":"2024-09-12T09:04:57.263Z","logger":"controllers.DatadogAgent","msg":"Reconciling DatadogAgent","datadogagent":{"name":"datadog","namespace":"system"}}
{"level":"INFO","ts":"2024-09-12T09:04:57.264Z","logger":"controllers.DatadogAgent","msg":"adding custom env var [{foo bar nil}]","datadogagent":{"name":"datadog","namespace":"system"}}
{"level":"INFO","ts":"2024-09-12T09:04:57.264Z","logger":"controllers.DatadogAgent","msg":"adding custom env var [{foo bar nil}]","datadogagent":{"name":"datadog","namespace":"system"}}
{"level":"INFO","ts":"2024-09-12T09:04:57.265Z","logger":"controllers.DatadogAgent","msg":"adding custom env var [{foo bar nil}]","datadogagent":{"name":"datadog","namespace":"system"}}
```